src/qibo/backends/numpy.py

```diff
@@ -72,14 +74,17 @@ def set_threads(self, nthreads):
     def cast(self, x, dtype=None, copy=False):
         if dtype is None:
             dtype = self.dtype
+
         if isinstance(x, self.tensor_types):
             return x.astype(dtype, copy=copy)
-        elif self.is_sparse(x):
+
+        if self.is_sparse(x):
             return x.astype(dtype, copy=copy)
+
         return np.asarray(x, dtype=dtype, copy=copy if copy else None)
 
     def is_sparse(self, x):
-        from scipy import sparse
+        from scipy import sparse  # pylint: disable=import-outside-toplevel
```

Comment:
Is this import needed? It seems to me that `sparse` is already imported top level in this module.

Reply:
True.